### DISCUSSION TITLE
Does @simplewebauthn/browser support Android browser?

### DISCUSSION THREAD

**Discussion by baoweili** [MasterKale/SimpleWebAuthn#607] [Troubleshooting]
When I use startRegistration function in the Android browser, it prompts "WebAuthn is not supported in this browser". Because it is judged by window.PublicKeyCredential, I checked the compatibility on **can i use** and found that window.PublicKeyCredential is not compatible with the Android browser. What should I do?

  @simplewebauthn/browser: "^10.0.0"

**Comment by MasterKale** [comment]
Hello @baoweili, if SimpleWebAuthn is telling you WebAuthn isn't supported, and if you manually checked `window.PublicKeyCredential` and saw that it's missing, then there's nothing you can do to support WebAuthn there. The solution is to switch to a different browser.

You mention "Android browser", which browser is that specifically?

**Reply by baoweili** [reply]
Thanks for your reply. Android browser means the browser that comes with Android phones. 

The reason I have this doubt is that the report from "can i use" shows that [webauthn](https://caniuse.com/?search=webauthn) supports some Android browsers, but [window.PublicKeyCredential](https://caniuse.com/?search=PublicKeyCredential) does not support it. I did an experiment and found that if [@passwordless-id/webauthn](https://github.com/passwordless-id/webauthn) is used in the Android browser, webauthn registration and authentication can be completed without judging window.PublicKeyCredential. So I wonder whether the judgment of window.PublicKeyCredential is necessary or whether there are other ways to be compatible with this scenario.

Thanks again for your reply!

**Reply by MasterKale** [reply]
Interesting, if you can can you report back on what the value of `typeof window.PublicKeyCredential` is in that Android browser? My library checks that it returns `'function'` but @passwordless-id/webauthn doesn't; that could explain why their library works but mine doesn't.

I looked around at a few other suggested ways to check for WebAuthn availability and @simplewebauthn/browser might be too strict...

**Reply by baoweili** [reply]
Thanks for your reply. I found that the value of `typeof window.PublicKeyCredential` is `undefined` in my Android browser.

Yes, @passwordless-id/webauthn is not checked in the registration process. I wonder if we can avoid actively checking `browserSupportsWebAuthn` in `startRegistration`, and let users in need manually call `browserSupportsWebAuthn` for judgment; Or whether we can explore whether it is compatible with the usage scenarios of Android browsers, because it is feasible when `window.PublicKeyCredential` check is not performed.
